### ISSUE TITLE
Widget interaction and native selections

### ISSUE THREAD

**Issue by mattleff** [liferay/alloy-editor#355]
I have an issue similar to #352. I'm attempting to display a toolbar when clicking on a widget. The change in #352 partly resolves my issue but I'm finding that the changes of db977d142b97d61f9026383a4b3f4a842b4b62ff are preventing the `editorInteraction` event from being fired when a widget is clicked. I've attempted to fire the `editorInteraction` event `.on('focus', ...` from my widget code but this introduces other issues with `selectionData`. I'm wondering if there's a better way to deal with widget interaction than commenting [this](https://github.com/liferay/alloy-editor/blob/master/src/core/selection-region.js#L159-L161) (which does work :stuck_out_tongue_winking_eye:).

**Comment by ipeychev** [comment]
Hey @mattleff,

With @dpobel we were discussing a possible solution for #352 and there is a patch in progress. It might make sense to check it and give some feedback.

Thanks,

**Comment by dpobel** [comment]
Hi @mattleff,

I had the exact same needs and followed more the less the same path as you :)

So in the end, I produced the patch in https://github.com/liferay/alloy-editor/pull/352 so that the styles toolbar is not restricted to content editable elements. In addition, as you said, you still have to transform the widget `focus` event into an `editorInteraction` event. I do that [my custom widget plugin](https://github.com/ezsystems/PlatformUIBundle/pull/359/files#diff-96a1b661c9a7efa9561bf708ebb1a542R41) this works pretty well.

Cheers
--
Damien

**Comment by mattleff** [comment]
@ipeychev @dpobel Thanks for the suggestions! I'm using the patch in #352, and I tried transforming the widget `focus` event into the `editorInteraction` event like you suggest @dpobel. That works ok, but it's hard to fake the `selectionData`. I tried:

```js
widget.on( "focus", function() {
	...
	editor.fire('editorInteraction', {
		nativeEvent: event,
		selectionData: editor.getSelectionData(),
	});
});
```

but because of db977d1 `editor.getSelectionData()` returns `null`. I tried:

```js
...
editor.fire('editorInteraction', {
	nativeEvent: event,
	selectionData: {},
});
```

but that causes issues with selection tests. For example the [image selection test](https://github.com/liferay/alloy-editor/blob/master/src/ui/react/src/selections/selection-test.js#L13) requires `selectionData` to have an `.element`. I could either rewrite the selection tests to not require `selectionData` or revert db977d1. Since we have ba45a6f5cfc865c6297ad7f30acde03a145d1cbc do we need db977d1 or could it be reverted?
